tests/Test/Xrefcheck/Util.hs

```diff
@@ -18,7 +21,30 @@ parse :: Flavor -> ScanAction
 parse fl path =
   markdownScanner MarkdownConfig { mcFlavor = fl } path
 
-mockServer :: IO ()
-mockServer = run 3000 $ do
+mockServerUrl :: MockServerPort -> Text -> Text
+mockServerUrl (MockServerPort port) s = toText ("http://127.0.0.1:" <> show port <> s)
+
+mockServer :: MockServerPort -> IO ()
+mockServer (MockServerPort port) = run port $ do
   route "/401" $ pure $ toResponse ("" :: Text, unauthorized401)
   route "/403" $ pure $ toResponse ("" :: Text, forbidden403)
+
+-- | All options needed to configure the mock server.
+mockServerOptions :: [OptionDescription]
+mockServerOptions =
+  [ Tasty.Option (Proxy @MockServerPort)
+  ]
+
+-- | Option specifying FTP host.
+newtype MockServerPort = MockServerPort Int
+  deriving stock (Show, Eq)
+
+instance IsOption MockServerPort where
+  defaultValue = MockServerPort 3000
+  optionName = "mock-server-port"
+  optionHelp = "[Test.Xrefcheck.Util] Mock server port"
+  parseValue v = MockServerPort <$> safeRead v
+  optionCLParser = MockServerPort <$> option auto
+    (  long (untag (optionName :: Tagged MockServerPort String))
```

Comment:
Can't you just use `optionName @MockServerPort` here instead of `Tagged`?

Reply:
I bet I can, but I followed the pattern established in `ftp-tests/Test/Xrefcheck/FtpLinks.hs`.

Will remove `Tagged` in both places then.

Reply:
Fixed

Reply:
I asked because I noticed a new dependency being added which seems unnecessary (I mean `tagged`). Turns out it's not really new, it's new for that test suite, but not for the whole package. However, I don't quite understand the new code:
```
  optionCLParser = FtpHostOpt <$> strOption
    (  long (untag @FtpHostOpt optionName)
    <> help (untag @FtpHostOpt optionHelp)
    )
```

Can't it be just
```
  optionCLParser = FtpHostOpt <$> strOption
    (  long optionName
    <> help optionHelp
    )
```
?
We have 
> long :: [HasName](https://hackage.haskell.org/package/optparse-applicative-0.18.1.0/docs/Options-Applicative-Builder.html#t:HasName) f => [String](https://hackage.haskell.org/package/base-4.16.3.0/docs/Data-String.html#t:String) -> [Mod](https://hackage.haskell.org/package/optparse-applicative-0.18.1.0/docs/Options-Applicative-Builder.html#t:Mod) f a

and 

> strOption :: [IsString](https://hackage.haskell.org/package/base-4.16.3.0/docs/Data-String.html#t:IsString) s => [Mod](https://hackage.haskell.org/package/optparse-applicative-0.18.1.0/docs/Options-Applicative-Builder.html#t:Mod) [OptionFields](https://hackage.haskell.org/package/optparse-applicative-0.18.1.0/docs/Options-Applicative-Builder.html#t:OptionFields) s -> [Parser](https://hackage.haskell.org/package/optparse-applicative-0.18.1.0/docs/Options-Applicative-Types.html#t:Parser) s

and we want the result of `strOption (…)` to be `Parser Text`. I think my short version should just work, because the argument of the `FtpHostOpt` constructor is `Text`, but if it doesn't, I think it can be solved by explicit type applications for `long` or `help`.

Reply:
The problem is that `optionName :: Tagged v String` and `optionHelp :: Tagged v String` and we need to unwrap them to get the strings. See the definition of [IsOption](https://hackage.haskell.org/package/tasty-1.5/docs/Test-Tasty-Options.html#t:IsOption) in `Test.Tasty.Options`:

```
class Typeable v => IsOption v where
  defaultValue :: v
  parseValue :: String -> Maybe v
  optionName :: Tagged v String
  optionHelp :: Tagged v String
  showDefaultValue :: v -> Maybe String
  optionCLParser :: Parser v
```

So we can't fully get rid of `Tagged` there. It's used to avoid ambiguous type variables.

Regarding a new dependency, it's not entirely new: I had to make it explicit to use `untag`, but we already depended on `tagged` transitively via `tasty`.

Reply:
Oh, ok, I see now. For some reason I thought that `optionName` was a local variable, not a type class member. Okay, then it all makes sense to me.